docs/basic/merge-sort.md

```diff
@@ -107,9 +116,52 @@ def merge_sort(ll, rr):
 
 归并排序还可以用来求逆序对的个数。
 
-所谓逆序对，就是对于一个数组 $a$，满足 $a_{i} > a_{j}$ 且 $i < j$ 的数对 $(i, j)$。
+所谓逆序对，就是对于一个数组 $a$，满足 $a_{i} > a_{j}$ 且 $i < j$ 的数对 $(i, j)$, 那么如何求逆序对呢? 
 
-代码实现中注释掉的 `ans += mid - p` 就是在统计逆序对个数。具体来说，算法把靠后的数放到前面了（较小的数放在前面），所以在这个数原来位置之前的、比它大的数都会和它形成逆序对，而这个个数就是还没有合并进去的数的个数，即 `mid - p`。
+1. 暴力枚举起始点和终止点。时间复杂度 $O(n^2)$，显然不是我们所想要的。
+2. 考虑利用归并排序的性质，把靠后的数放到前面了（较小的数放在前面），所以在这个数原来位置之前的、比它大的数都会和它形成逆序对，而这个个数就是还没有合并进去的数的个数，即 `mid - p + 1`。请看上面的归并排序程序, 当待合并的数组为：
+```
+x = {3, 6}
+y = {1}
+```
+的时候，$3$ 和 $6$ 都会被放在 $1$ 的前面，于是可以得出这轮合并中形成了 2 个逆序对。
+
+下面给出求逆序对数量的模板：
+??? note "[洛谷P1908 - 逆序对](https://www.luogu.com.cn/problem/P1908)"
```

Comment:
```suggestion
??? note "[Luogu P1908 逆序对](https://www.luogu.com.cn/problem/P1908)"
```

formats in other prs and master are not like this, so please change?

Reply:
ok